python/strings_udf/strings_udf/lowering.py

```diff
@@ -0,0 +1,637 @@
+    result = context.compile_internal(
+        builder,
+        call_string_view_lt,
+        nb_signature(
+            types.boolean,
+            types.CPointer(string_view),
+            types.CPointer(string_view),
+        ),
+        (sv_ptr, rhs_ptr),
+    )
+
+    return result
+
+
+# read-only functions
+# We will provide only one overload for this set of functions, which will
+# expect a string_view. When a literal is encountered, numba will promote it to
+# a string_view whereas when a dstring is encountered, numba will convert it to
+# a view via its native view() method.
+
+# casts
+@cuda_lowering_registry.lower_cast(types.StringLiteral, string_view)
+def cast_string_literal_to_string_view(context, builder, fromty, toty, val):
+    """
+    Cast a literal to a string_view
+    """
+    # create an empty string_view
+    sv = cgutils.create_struct_proxy(string_view)(context, builder)
+
+    # set the empty strview data pointer to point to the literal value
+    s = context.insert_const_string(builder.module, fromty.literal_value)
+    sv.data = context.insert_addrspace_conv(
+        builder, s, nvvm.ADDRSPACE_CONSTANT
+    )
+    sv.length = context.get_constant(types.int32, len(fromty.literal_value))
+    sv.bytes = context.get_constant(
+        types.int32, len(fromty.literal_value.encode("UTF-8"))
+    )
+
+    return sv._getvalue()
+
+
+def call_string_view_startswith(sv, substr):
+    return _string_view_startswith(sv, substr)
+
+
+@cuda_lower("StringView.startswith", string_view, string_view)
+def string_view_startswith_impl(context, builder, sig, args):
+    sv_ptr, substr_ptr = builder.alloca(args[0].type), builder.alloca(
+        args[1].type
+    )
+
+    builder.store(args[0], sv_ptr)
+    builder.store(args[1], substr_ptr)
+
+    result = context.compile_internal(
+        builder,
+        call_string_view_startswith,
+        nb_signature(
+            types.boolean,
+            types.CPointer(string_view),
+            types.CPointer(string_view),
+        ),
+        (sv_ptr, substr_ptr),
+    )
+
+    return result
+
+
+def call_string_view_endswith(sv, substr):
+    return _string_view_endswith(sv, substr)
+
+
+@cuda_lower("StringView.endswith", string_view, string_view)
+def string_view_endswith_impl(context, builder, sig, args):
+    sv_ptr, substr_ptr = builder.alloca(args[0].type), builder.alloca(
+        args[1].type
+    )
+
+    builder.store(args[0], sv_ptr)
+    builder.store(args[1], substr_ptr)
+
+    result = context.compile_internal(
+        builder,
+        call_string_view_endswith,
+        nb_signature(
+            types.boolean,
+            types.CPointer(string_view),
+            types.CPointer(string_view),
+        ),
+        (sv_ptr, substr_ptr),
+    )
+
+    return result
+
+
+def call_string_view_count(st, substr):
+    return _string_view_count(st, substr)
+
+
+@cuda_lower("StringView.count", string_view, string_view)
+def string_view_coount_impl(context, builder, sig, args):
```

Comment:
Is this typo in a function name hiding something not working as expected? I don't see where this function is used, do we need bindings / tests for it?
```suggestion
def string_view_count_impl(context, builder, sig, args):
```

Reply:
This should be corrected, but luckily it's not masking anything that isn't working. For these lowering implementations, numba doesn't care what the function name is - it only cares that an implementation accepting to `string_view` objects is tied to `StringView`'s `count` attribute. 

That said, you're right about there being missing tests on the cuDF side for this! I apparently added tests in the `strings_udf` typing area that test the scalar string `count` attribute, but there's no equivalent tests in cuDF for `MaskedType(string_view)`. So I added them.